### PR TITLE
python3Packages.qiskit: 0.24.1 -> 0.25.0

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Motivation for this change

https://github.com/Qiskit/qiskit/releases/tag/0.25.0.

* Adds ``tweedledum``, an optional-ish dependency of ``qiskit-terra``
* Updates ``pylatexenc``

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/tweedledum/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, cmake
+, ninja
+, scikit-build
+  # Check Inputs
+, pytestCheckHook
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/tweedledum/default.nix:8]:
```suggestion
, scikit-build
, pytestCheckHook
```

File: pkgs/development/python-modules/tweedledum/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  nativeBuildInputs = [ cmake ninja scikit-build ];
+  dontUseCmakeConfigure = true;
+
+  pythonImportsCheck = [ "tweedledum" ];
+
+  # TODO: use pytest, but had issues with finding the correct directories
+  checkPhase = ''
+    python -m unittest discover -s ./python/test -t .
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/tweedledum/default.nix:31]: set pytestCheckFlags to ``python/test``.

File: pkgs/development/python-modules/qiskit-aer/default.nix [hunk 3]
```diff
@@ -62,7 +62,12 @@ buildPythonPackage rec {
...
 
   postPatch = ''
-    substituteInPlace setup.py --replace "'cmake!=3.17,!=3.17.0'," ""
+    substituteInPlace setup.py \
+      --replace "'cmake!=3.17,!=3.17.0'," "" \
+      --replace "'pybind11', min_version='2.6'" "'pybind11'" \
+      --replace "pybind11>=2.6" "pybind11" \
+      --replace "scikit-build>=0.11.0" "scikit-build" \
+      --replace "min_version='0.11.0'" ""
   '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/qiskit-aer/default.nix:71]: Why got those dependencies removed?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/qiskit-aer/default.nix (hunk 3 above)
Comment (drewrisinger) [reply pkgs/development/python-modules/qiskit-aer/default.nix:71]: This is a carry-over from my NUR build, where I support nixos [20.03, 20.09]. The min versions don't apply here b/c the ``nixpkgs`` versions are higher than the min requirements. Good spot.

File: pkgs/development/python-modules/tweedledum/default.nix (hunk 2 above)
Comment (drewrisinger) [reply pkgs/development/python-modules/tweedledum/default.nix:31]: With the suggested change:
```
________________ ERROR collecting python/test/test_BitVector.py ________________
ImportError while importing test module '/build/source/python/test/test_BitVector.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/5470xw15wnn972ap0c4f7q642z4nvh6f-python3-3.8.8/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
python/test/test_BitVector.py:10: in <module>
    from tweedledum.bool_function_compiler import BitVec
python/tweedledum/__init__.py:5: in <module>
    from ._tweedledum import *
E   ModuleNotFoundError: No module named 'tweedledum._tweedledum'
______________ ERROR collecting python/test/test_BoolFunction.py _______________
ImportError while importing test module '/build/source/python/test/test_BoolFunction.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/5470xw15wnn972ap0c4f7q642z4nvh6f-python3-3.8.8/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
python/test/test_BoolFunction.py:7: in <module>
    from tweedledum.bool_function_compiler import BitVec, BoolFunction
python/tweedledum/__init__.py:5: in <module>
    from ._tweedledum import *
E   ModuleNotFoundError: No module named 'tweedledum._tweedledum'
_________________ ERROR collecting python/test/test_Circuit.py _________________
ImportError while importing test module '/build/source/python/test/test_Circuit.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/5470xw15wnn972ap0c4f7q642z4nvh6f-python3-3.8.8/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
python/test/test_Circuit.py:7: in <module>
    from tweedledum.ir import Circuit
python/tweedledum/__init__.py:5: in <module>
    from ._tweedledum import *
E   ModuleNotFoundError: No module named 'tweedledum._tweedledum'
=========================== short test summary info ============================
ERROR python/test/test_BitVector.py
ERROR python/test/test_BoolFunction.py
ERROR python/test/test_Circuit.py
!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!
============================== 3 errors in 0.37s ===============================
builder for '/nix/store/m90fn5fjd3wld2j1hqnky7j115xhn1cx-python3.8-tweedledum-1.0.0.drv' failed with exit code 2
```

Comment (SuperSandro2000) [reply pkgs/development/python-modules/tweedledum/default.nix:29]: Seems like it the package won't work with pytest.

**Comment by SuperSandro2000** [comment]
```
python3.8-qiskit-aqua> Requirement already satisfied: numpy<=1.20.1,>=1.17 in /nix/store/pwjsgyqgsvqi5wg5h7rqk57i9qv9lqyc-python3.8-numpy-1.20.1/lib/python3.8/site-packages (from qiskit-aqua==0.9.0) (1.20.1)
python3.8-qiskit-aqua> Requirement already satisfied: yfinance<=0.1.55 in /nix/store/rl858rjjm13385hs9v5ksh3nhkcvpiai-python3.8-yfinance-0.1.55/lib/python3.8/site-packages (from qiskit-aqua==0.9.0) (0.1.55)
python3.8-qiskit-aqua> ERROR: Could not find a version that satisfies the requirement h5py<=3.1.0 (from qiskit-aqua)
python3.8-qiskit-aqua> ERROR: No matching distribution found for h5py<=3.1.0
```

**Comment by drewrisinger** [comment]
>
> ```
> python3.8-qiskit-aqua> Requirement already satisfied: numpy<=1.20.1,>=1.17 in /nix/store/pwjsgyqgsvqi5wg5h7rqk57i9qv9lqyc-python3.8-numpy-1.20.1/lib/python3.8/site-packages (from qiskit-aqua==0.9.0) (1.20.1)
> python3.8-qiskit-aqua> Requirement already satisfied: yfinance<=0.1.55 in /nix/store/rl858rjjm13385hs9v5ksh3nhkcvpiai-python3.8-yfinance-0.1.55/lib/python3.8/site-packages (from qiskit-aqua==0.9.0) (0.1.55)
> python3.8-qiskit-aqua> ERROR: Could not find a version that satisfies the requirement h5py<=3.1.0 (from qiskit-aqua)
> python3.8-qiskit-aqua> ERROR: No matching distribution found for h5py<=3.1.0
> ```

This should be fixed now. I hadn't realized that ``qiskit-aqua`` had pinned all of its dependencies to <= the current versions, which obviously doesn't work with ``nixpkgs``'s rolling releases. Fixed by unpinning all requirements in ``requirements.txt``.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118729` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
18 packages built:
</summary>
<ul>
<li>python38Packages.Quandl (python38Packages.quandl)</li>
<li>python38Packages.pylatexenc</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python38Packages.qiskit-terra</li>
<li>python38Packages.tweedledum</li>
<li>python39Packages.Quandl (python39Packages.quandl)</li>
<li>python39Packages.pylatexenc</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ibmq-provider</li>
<li>python39Packages.qiskit-ignis</li>
<li>python39Packages.qiskit-terra</li>
<li>python39Packages.tweedledum</li>
</ul>
</details>

**Pull request merged**